### ISSUE TITLE
[Feature] InjectorService, add manually a service (allready constructed).

### ISSUE THREAD

**Issue by Romakita** [Romakita/ts-express-decorators#36]

**Comment by Romakita** [comment]
InjectorService can use a service built manually and/or not typed.

Here's how it works:
```typescript
// fooservice.ts
import {InjectorService} from "ts-express-decorators";

export interface IMyFooService {
   getFoo(): string;
}

export type MyFooService = IMyFooService;
export const MyFooService = Function;

InjectorService.factory(MyFooService, {
     getFoo:  () => "test"
});

// otherservice.ts
import {MyFooService} from "./fooservice.ts";

@Service()
export default class OtherService {
     constructor(myFooService: MyFooService){
           console.log(myFooService.getFoo()); /// "test"
     }
}
```

This feature will be released for the v1.4.0-1 (beta) (soon)

**Comment by m0uneer** [comment]
This example is still valid and working?
I tried it and got:

> Process finished with exit code 0

I don't know how to set debugging mode (And I would be grateful if you advice) so I dived into the module to find this error being thrown:

> TypeError: _this.expressApplication.use is not a function
>     at /home/mouneer/djaty/djaty-server/node_modules/ts-express-decorators/lib/services/controller.js:136:50
>     at Array.forEach (native)

I use 

> "ts-express-decorators": "^1.4.0-20",

Any thoughts?

**Comment by Romakita** [comment]
Hello @m0uneer,

Thanks for your issue because you found a big fail with the InjectorService.

Effectively the example does not work. I've only test with ExpressApplication as Factory but I never test with other factory. My mistake.

So I've fix the problem with the 1.4.0-21 and the right example is the following:

```typescript
// fooservice.ts
import {InjectorService} from "ts-express-decorators";

export interface IMyFooService {
   getFoo(): string;
}

export type MyFooService = IMyFooService;
export const MyFooService = Symbol("MyFooService");

InjectorService.factory(MyFooService, {
     getFoo:  () => "test"
});

// otherservice.ts
import {MyFooService} from "./fooservice.ts";

@Service()
export default class OtherService {
     constructor(@Inject(MyFooService) myFooService: MyFooService){
           console.log(myFooService.getFoo()); /// "test"
     }
}
```
> Note: For a factory you must use the Inject decorator. I haven't found a solution without decorator :/

**Comment by m0uneer** [comment]
Great, this works. Thanks a lot.

Regarding the Inject decorator, why don't you read the function string, parse it then inject the required dependency?
